### PR TITLE
#163388248 validation messages for signup

### PR DESCRIPTION
**Pull request by unah254**
#### What does this PR do?
Have validation messages for signup/registration
#### Description of Task to be completed?
- add a test for a valid username
- add a test for alphanumeric password
- add serializer for a valid username
- add serializer for an alphanumeric password
#### How should this be manually tested?

- clone the repo and `cd` into it. 

- checkout to the branch
`git checkout ft-descriptive-validation-messages-163388248`

- Run server 
`python manage.py runserver`

- test on postman using the url `http://127.0.0.1:8000/api/users/`

- On the payload, input username, email and password i.e 
```
         {
		"username": "77877",
		"email": "meffdsfgfdf@gmail.com",
		"password": "Sanjihydfsvs$$$#34!"
	}
```
- Messages should appear i.e `Username cannot contain numbers only` and `Password must contain a number, capital letter and specuial charachter`

#### Any background context you want to provide?
None
#### What are the relevant pivotal tracker stories?
[#163388248](https://www.pivotaltracker.com/n/projects/2240063)

#### Screenshots
image

image

#### Questions:
How can I test for a serializer?

### CODE REVIEW

**Comment by TooColline** [comment]
@unah254 Good work, the messages are descriptive. However, reduce the maximum length of the password to something like 30 and fix the flake8 errors.

**Review by bryan-munene — COMMENTED** [review]
Hi, @unah254 This is nice work on the messages. However, your checks are failing especially the style tests by flake8. Solve those and I think you should be good to go.

**Comment by calebrotich** [comment]
You have some typos in the file named `test_signuup.py`. Line `100`. You have mispelled the words `Special characters`. Otherwise lgtm.

File: authors/apps/authentication/serializers.py [hunk 1]
```diff
@@ -16,6 +18,22 @@ class RegistrationSerializer(serializers.ModelSerializer):
         write_only=True
     )
 
+    def validate_username(self, data):
```
Comment (sam-karis) [inline authors/apps/authentication/serializers.py:21]: Hi @unah254 good work but I think this can be reduced to one regex.

**Review by sam-karis — CHANGES_REQUESTED** [review]

File: authors/apps/authentication/serializers.py (hunk 1 above)
Comment (unah254) [reply authors/apps/authentication/serializers.py:21]: Hello, thanks let me change

**Review by sam-karis — APPROVED** [review]
LGTM

**Pull request merged**